### ISSUE TITLE
CA1508 reported in catch block

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn-analyzers#6530]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2013): `Remove or refactor the condition(s) to avoid dead code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 7 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.102](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

False positive: "Remove or refactor the condition(s) to avoid dead code".

### Steps To Reproduce

````cs
ILogger? logger = null;
try {
  somethingThatCouldFail();
  logger = buildLogger();
  somethingElse();
}
catch (Exception e) {
  if (logger != null)                          // <------ 'logger != null' is always 'true'. Remove or refactor the condition(s) to avoid dead code. [MyProject]csharp(CA1508)
    logger.LogError(e, "Something failed");
  else
    Console.WriteLine("Something failed: " + e.Message);
}
````

### Expected behavior

No CA1508

### Actual behavior

CA1508

**Comment by koenigst** [comment]
A very similar case reports a false positive for `finally` blocks:

```csharp
var logger = buildLogger();
try
{
    somethingThatCouldFail();
    logger = null;
}
finally
{
    logger?.LogError("Something failed"); // <------ 'logger' is always 'null'. Remove or refactor the condition(s) to avoid dead code.
}
```

**Comment by mavasani** [comment]
> A very similar case reports a false positive for `finally` blocks:
> 
> ```cs
> var logger = buildLogger();
> try
> {
>     somethingThatCouldFail();
>     logger = null;
> }
> finally
> {
>     logger?.LogError("Something failed"); // <------ 'logger' is always 'null'. Remove or refactor the condition(s) to avoid dead code.
> }
> ```

@koenigst can you please also paste the contents of the method called inside the try block? Note that we do interprocedural analysis, and if we detect the code within “somethingThatCouldFail” is exception safe, then this may be by design. Otherwise, it would be a bug

**Comment by koenigst** [comment]
> @koenigst can you please also paste the contents of the method called inside the try block? Note that we do interprocedural analysis, and if we detect the code within “somethingThatCouldFail” is exception safe, then this may be by design. Otherwise, it would be a bug

@mavasani I updated my original comment. The method `somethingThatCouldFail` always throws.

**Comment by mavasani** [comment]
Verified this is fixed with #6828 and should be part of the next .NET 8 preview SDK
